packages/edit-navigation/src/store/test/menu-items-to-blocks.js

```diff
@@ -0,0 +1,387 @@
+/**
+ * Internal dependencies
+ */
+import menuItemsToBlocks from '../menu-items-to-blocks';
+
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import {
+	metadata,
+	settings,
+	name,
+} from '../../../../block-library/src/navigation-link/index';
```

Comment:
Not sure how I feel about this import, but since `block-library` doesn't export individual blocks, I guess this is the only option.

Reply:
I know, it's not great 😞  

I was considering mocking out `createBlock`. @kevin940726 has some experience with mocking such functions I think. I'd be interested in his thoughts.

Reply:
> I was considering mocking out `createBlock`

Yeah I'd decouple that since we're not interested in block internals for this instance. We can sometimes change signatures to make testing/mocking easier. See example in 

https://github.com/WordPress/gutenberg/blob/c8c463c62241c0854097cf71ce583adf920ed084/packages/block-library/src/page-list/test/convert-to-links-modal.js#L92-L97

Reply:
I was able to find this snippet in tests. I can't say I fully understand how it works.

https://github.com/WordPress/gutenberg/blob/f1f63ab22da1994ab2a71056e092c8900f12467c/packages/edit-navigation/src/store/test/resolvers.js#L13-L29

Reply:
> Yeah I'd decouple that since we're not interested in block internals for this instance. 

I agree. 

> I was able to find this snippet in tests. 

Thanks this saved me some time.


> I can't say I fully understand how it works.

It works by creating a Jest module mock which is cool as it allows you to mock things that aren't directly under test but which are called by the code under test without having to do complex dependency injection.

```
const blocks = jest.requireActual( '@wordpress/blocks' );
```

This requires the _real_ module code for `@wordpress/blocks`.

But then...

```
return {
		...blocks,
		createBlock( name, attributes, innerBlocks ) {
			return {
				name,
				attributes,
				innerBlocks,
			};
		},
	};
```

...we return the following from our mocked version of `@wordpress/blocks` which is basically all the APIs from the module but with a customised (mocked!) version of `createBlock` instead of the original one.

This means when our code under test calls out to `createBlock` (which it does) then our mocked version of `createBlock` will be called. As long as the mock has the same return signature as the real module then we're all good.

Reply:
I updated the tests. I was also able to remove the code which was stripping out the `clientId`s which tidied things up.